src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -35,17 +36,21 @@ private void ExecuteForegroundAction(Action action)
             {
                 lock (_queueGate)
                 {
-                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWith(
-                        _ =>
+                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWithFromAsync(
+                        async _ =>
                         {
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true);
+
                             // since execution is now technically asynchronous
                             // only execute action if project is not disconnected and currently being tracked.
                             if (!_disconnected && this.ProjectTracker.ContainsProject(this))
                             {
                                 action();
                             }
                         },
-                        ForegroundTaskScheduler);
+                        CancellationToken.None,
+                        TaskContinuationOptions.ExecuteSynchronously,
```

Comment:
different options than usual.  intentional?

Reply:
➡️ Situation is most similar to https://github.com/dotnet/roslyn/pull/28781#discussion_r208411013, but has a dedicated follow-up item due to https://github.com/dotnet/roslyn/pull/28781#discussion_r208393997